src/main/java/com/cloudogu/scm/smeagol/search/IndexListener.java

```diff
@@ -81,6 +82,22 @@ public void handle(DefaultBranchChangedEvent event) {
     submit(event.getRepository());
   }
 
+  @Subscribe
+  public void handle(ReindexRepositoryEvent event) {
+    Repository repository = event.getRepository();
+    LOG.debug(
+      "received reindex event for repository {}, delete and recreate index from scratch",
+      repository
+    );
+    if ("git".equals(repository.getType())) {
+      searchEngine.forType(SmeagolDocument.class)
+        .forResource(repository)
+        .update(new ReindexTask(repository));
```

Comment:
??? This can't be right

Reply:
All the other event listeners skip non-git repositories, why would this event be different ?

Reply:
You are right. This feels awkward but makes sense for this plugin, since Smeagol is git based.

You could use SmeagolRepositoryFilter.